storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RetriableRemoteStorageException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.log.remote.storage;
+
+/**
+ * Represents an exception that indicates a retriable error occurred during remote storage operations.
+ * This exception is thrown when an operation against a remote storage system has failed due to transient
+ * or temporary issues, and the operation has a reasonable chance of succeeding if retried.
+ */
+public class RetriableRemoteStorageException extends RemoteStorageException {
```

Comment:
I think we should make the exception name clear for troubleshooting. Although we already include the reason in the error message, sometimes we only got exception class name without messages. Maybe we can keep this class and create another exception like `RemoteStorageNotReadyException` inherited from `RetriableRemoteStorageException`, WDYT?

Reply:
Added another RemoteStorageNotReadyException.